python/hopsworks_common/core/search_api.py

```diff
@@ -0,0 +1,304 @@
+
+        ```
+        # Arguments
+            search_term: the term to search for.
+            filter_by: filter results by a specific field (default is None).
+            offset: the number of results to skip (default is 0).
+            limit: the number of search results to return (default is 100).
+        # Returns
+            `FeaturestoreSearchResult`: The results.
+        # Raises
+            `ValueError`: If the search term is not provided or if the filter_by is not one of the allowed values.
+            `hopsworks.client.exceptions.RestAPIError`: If the backend encounters an error when handling the request
+        """
+        return self._search(
+            search_term, "ALL", filter_by=filter_by, offset=offset, limit=limit
+        )
+
+    # FEATUREGROUP
+    def featuregroup_search(
+        self,
+        search_term: str,
+        filter_by: FILTER_BY_ARG = None,
+        offset: int = 0,
+        limit: int = 100,
+    ):
+        """
+        Search for feature group.
+        ```python
+
+        import hopsworks
+
+        project = hopsworks.login()
+
+        search_api = project.get_search_api()
+
+        result = search_api.featuregroup_search("search_term")
+
+        # get feature group instance
+        featuregroup = result.featuregroups[0].get_feature_group()
+
+        ```
+        # Arguments
+            search_term: the term to search for.
+            filter_by: filter results by a specific field (default is None).
+            offset: the number of results to skip (default is 0).
+            limit: the number of search results to return (default is 100).
+        # Returns
+            `FeatureGroupSearchResult`: The results.
+        # Raises
+            `ValueError`: If the search term is not provided or if the filter_by is not one of the allowed values.
+            `hopsworks.client.exceptions.RestAPIError`: If the backend encounters an error when handling the request
+        """
+        return search_results.FeatureGroupSearchResult(
+            self._search(
+                search_term,
+                "FEATUREGROUP",
+                filter_by=filter_by,
+                offset=offset,
+                limit=limit,
+            )
+        )
+
+    # FEATUREVIEW
+    def featureview_search(
+        self,
+        search_term: str,
+        filter_by: FILTER_BY_ARG = None,
+        offset: int = 0,
+        limit: int = 100,
+    ):
+        """
+        Search for feature views.
+        ```python
+
+        import hopsworks
+
+        project = hopsworks.login()
+
+        search_api = project.get_search_api()
+
+        result = search_api.featureview_search("search_term")
+
+        # get feature view instance
+        featureview = result.feature_views[0].get_feature_view()
+
+        ```
+        # Arguments
+            search_term: the term to search for.
+            filter_by: filter results by a specific field (default is None).
```

Comment:
I think we should mention what are the possible filter values in the docstring.

Reply:
@ErmiasG same as above